### PR TITLE
fixes core#580 - view all groups when appropriately permissioned

### PR DESCRIPTION
**Pull request by MegaphoneJon**
Overview
----------------------------------------
Given a user who:
* Doesn't have "View All Contacts" permission;
* DOES have an ACL that gives permission to Edit "All Groups";

The "Manage Groups" screen shows no groups.

Before
----------------------------------------
Selecting "Manage Groups" with an ACL to display "All groups" shows no groups.

After
----------------------------------------
Selecting "Manage Groups" with an ACL to display "All groups" shows all groups.

Technical Details
----------------------------------------
This happens because `CRM_ACL_BAO_ACL::group()` expects an argument `$allGroups` to be passed, with an array of all the groups for this particular context.  The API code path populates this argument; the UI code path doesn't.

Comments
----------------------------------------
This is the same approach as #13242 but placing the fix in a more appropriate location.

### CODE REVIEW

**Comment by seamuslee001** [comment]
Test failures look related here @MegaphoneJon

**Comment by seamuslee001** [comment]
The fails are 

```
CRM_Group_Page_AjaxTest.testGroupListAclGroupHookDisabled
CRM_Group_Page_AjaxTest.testGroupListAclGroupHookDisabledFound
CRM_Group_Page_AjaxTest.testGroupListAclGroupHookTitleFound
CRM_Group_Page_AjaxTest.testGroupListAclGroupHookAll
```

**Comment by MegaphoneJon** [comment]
OK, the issue was that `allGroup()` wasn't being told to include inactive groups.
I decided to switch to `buildOptions()` because `allGroup()` is deprecated.  This revealed a bug in `CRM_Contact_BAO_GroupContact::buildOptions()` that it doesn't take the `$props` argument.  It seems intentional - but no other usage of the function passes the third argument, so it looks safe to change.

Finally - this exposed an issue with how the action links are generated.  If your bitmask evaluates to 0, it shows all, rather than no action links.  So I wrapped that in a conditional to display properly so folks with fewer permissions don't see links they shouldn't.

File: tests/phpunit/CRM/Group/Page/AjaxTest.php [hunk 1]
```diff
@@ -650,4 +650,61 @@ public function hook_civicrm_aclGroup($type, $contactID, $tableName, &$allGroups
     }
   }
 
+  public function testEditAllGroupsACL() {
+    $this->setupEditAllGroupsACL();
+    $params = $this->_params;
+    $groups = CRM_Contact_BAO_Group::getGroupListSelector($params);
+    print_r($groups);
```
Comment (seamuslee001) [inline tests/phpunit/CRM/Group/Page/AjaxTest.php]: @MegaphoneJon do we need the print_r here?

Comment (colemanw) [reply tests/phpunit/CRM/Group/Page/AjaxTest.php]: @MegaphoneJon getting the integrated debugger going with PhpStorm is a pain to set up but so worth it!

**Comment by colemanw** [comment]
@MegaphoneJon I agree with you about these changes and passing `$props` in buildOptions. Thanks for adding tests. Good to merge.

**Pull request merged**